runtime/compiler/env/j9methodServer.cpp

```diff
@@ -189,8 +189,17 @@ TR_ResolvedJ9JITServerMethod::getClassFromConstantPool(TR::Compilation * comp, u
    if (cpIndex == -1)
       return NULL;
 
-   TR::CompilationInfoPerThread *compInfoPT = _fe->_compInfoPT;
+   auto compInfoPT = (TR::CompilationInfoPerThreadRemote *) _fe->_compInfoPT;
+   bool doRuntimeResolve = compInfoPT->getClientData()->getRtResolve() && !comp->ilGenRequest().details().isMethodHandleThunk();
+   if (doRuntimeResolve &&
+       performTransformation(comp, "Setting as unresolved class from CP cpIndex=%d\n",cpIndex))
       {
+      return NULL;
+      }
+
+      {
+      // This persistent cache must only be checked when doRuntimeResolve is false,
```

Comment:
Is the comment accurate? For a method handle thunk compilation we want to resolve (and thus check the cache) anyway.

Reply:
The comment talks about the variable `doRuntimeResolve` which includes checking for thunk compilation
```
bool doRuntimeResolve = compInfoPT->getClientData()->getRtResolve() && !comp->ilGenRequest().details().isMethodHandleThunk();
```

Reply:
Right, but I think it should have a negation:
"otherwise a resolved class cached during a NON method handle thunk compilation can be returned, instead of NULL"
because during method handle thunk compilations we always use the cache (in the old and in the new implementation)

Reply:
Oh, ok. Rephrased to make it more clear.